src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -1762,10 +1762,16 @@ public enum ServerCompatibilityMode
     /// </summary>
     Redshift,
     /// <summary>
-    /// The server is doesn't support full type loading from the PostgreSQL catalogs, support the basic set
+    /// The server doesn't support type loading from the PostgreSQL catalogs, support the basic set
     /// of types via information hardcoded inside Npgsql.
     /// </summary>
     NoTypeLoading,
+    /// <summary>
+    /// The server doesn't support full type loading from the PostgreSQL catalogs, or the application has special requirements/problems
+    /// regarding connection startup and no need to support custom types. Support all types with hardcoded oids via information inside
+    /// Npgsql that has been generated from reading PostgreSQL's .dat files.
+    /// </summary>
+    AllHardCodedTypes,
```

Comment:
Do we really need a separate mode in addition to NoTypeLoading? In other words, shouldn't this just replace NoTypeLoading?

Assuming so, we can remove PostgresTypeOIDs, [BuiltInPostgresType], etc.

Reply:
> Do we really need a separate mode in addition to NoTypeLoading?

No we probably don't. I opted for the conservative approach of adding a feature, testing it, deprecating the old one and thereafter removing it but if you're feeling courageous we can do all of it in one step.

Reply:
I think your new mode is supposed to subsume the old one - any type currently supported by NoTypeLoading should also be supported in your new implementation (if not, that's a bug we'd fix in a patch). As far as users are concerned, whether we hardcode OIDs or code-generate them from .dat is an implementation detail, so yeah, I'd be in favor of replacing the existing mechanism with your new one...